libminifi/src/core/json/JsonFlowSerializer.cpp

```diff
@@ -68,18 +68,42 @@ void JsonFlowSerializer::encryptSensitiveProperties(rapidjson::Value& property_j
   }
 }
 
-std::string JsonFlowSerializer::serialize(const core::ProcessGroup& process_group, const core::flow::FlowSchema& schema, const utils::crypto::EncryptionProvider& encryption_provider,
-    const std::unordered_map<utils::Identifier, core::flow::Overrides>& overrides) const {
-  gsl_Expects(schema.root_group.size() == 1 && schema.identifier.size() == 1 &&
-      schema.processors.size() == 1 && schema.processor_properties.size() == 1 &&
-      schema.controller_services.size() == 1 && schema.controller_service_properties.size() == 1);
+void JsonFlowSerializer::encryptSensitiveParameters(rapidjson::Value& flow_definition_json, rapidjson::Document::AllocatorType& alloc, const core::flow::FlowSchema& schema,
+    const utils::crypto::EncryptionProvider& encryption_provider, const std::unordered_map<utils::Identifier, core::flow::Overrides>& overrides) const {
+  if (!flow_definition_json.HasMember(schema.parameter_contexts[0])) {
+    return;
+  }
 
-  rapidjson::Document doc;
-  auto alloc = doc.GetAllocator();
-  rapidjson::Value flow_definition_json;
-  flow_definition_json.CopyFrom(flow_definition_json_, alloc);
-  auto& root_group = getMember(flow_definition_json, schema.root_group[0]);
+  auto parameter_contexts = getMember(flow_definition_json, schema.parameter_contexts[0]).GetArray();
+  for (auto& parameter_context : parameter_contexts) {
+    for (auto& parameter : getMember(parameter_context, schema.parameters[0]).GetArray()) {
+      bool is_sensitive = getMember(parameter, schema.sensitive[0]).GetBool();
+      if (!is_sensitive) {
+        continue;
+      }
+      const std::string parameter_context_id_str{getMember(parameter_context, schema.identifier[0]).GetString(), getMember(parameter_context, schema.identifier[0]).GetStringLength()};
+      const auto parameter_context_id = utils::Identifier::parse(parameter_context_id_str);
+      if (!parameter_context_id) {
+        logger_->log_warn("Invalid parameter context ID found in the flow definition: {}", parameter_context_id_str);
+        continue;
+      }
+
+      std::string parameter_value{getMember(parameter, schema.value[0]).GetString(), getMember(parameter, schema.value[0]).GetStringLength()};
+      if (overrides.contains(*parameter_context_id)) {
+        const auto& override_values = overrides.at(*parameter_context_id);
+        const auto parameter_name = getMember(parameter, schema.name[0]).GetString();
```

Comment:
we should use `GetStringLength()` here, too, as elsewhere (eg. at line 91 above)

Reply:
Good point, updated in 887357fa01a2aafb3269d29f9c52c682cdede37b